code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+		return
+
+	var/mob/living/carbon/human/punched = attacked
+	if(punched.mind && punched.stat != DEAD)
+		puncher.adjustBruteLoss(-20)//Heal the puncher per punch. Only works on players that are alive.
+		puncher.adjustFireLoss(-20)
+	punched.adjustCloneLoss(20) //genetic damage is the closest thing we have to soul damage
+	playsound(puncher, 'sound/effects/pop_expl.ogg', 75, TRUE)
+	to_chat(puncher, "<span class='warning'>You punch [punched]'s soul with all your might!</span>")
+	to_chat(punched, "<span class='warning'>You feel like something very important got punched out of you!</span>")
+	puncher.do_attack_animation(punched, ATTACK_EFFECT_SMASH)
+	return COMPONENT_NO_ATTACK_HAND //here to make sure we replace the original punch with our genetic punch
+
+
+///takes away the overlay and the genetic punch when you leave the rune
+/obj/effect/warped_rune/redspace/Uncrossed(atom/movable/crossing)
+	. = ..()
+	if(!istype(crossing,/mob/living/carbon/human))
+		return
+	var/mob/living/carbon/human/enraged = crossing
+	UnregisterSignal(enraged, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+	enraged.cut_overlay(hate_fire)
+
+///destroying the rune will also remove the punch force of the persons on the rune.
+/obj/effect/warped_rune/redspace/Destroy()
+	for(var/mob/living/carbon/human/enraged in rune_turf)
+		UnregisterSignal(enraged, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+		enraged.cut_overlay(hate_fire)
+	return ..()
+
+
+/* Green rune vores plasma and spews out xeno resin which lets you build xeno structure*/
+
+
+/obj/item/slimecross/warping/green
+	colour = "green"
+	runepath = /obj/effect/warped_rune/greenspace
+	drawing_time = 100
+
+
+/obj/effect/warped_rune/greenspace
+	icon_state = "xeno_rune"
+	desc = "We will build walls out of our fallen foes, they shall fear our very buildings."
+	max_cooldown = 100
+	///path of the xeno resin sheet to spawn by the rune
+	var/obj/item/stack/sheet/xeno_resin/resin
+
+
+/obj/item/stack/sheet/xeno_resin
+	name = "Resin sheets"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "nestoverlay" //literally just the xeno nest icon
+	merge_type = /obj/item/stack/sheet/xeno_resin
+
+
+/obj/effect/warped_rune/greenspace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/greenspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		transmute_resin()
+
+
+/obj/effect/warped_rune/greenspace/proc/transmute_resin()
+	for(var/obj/item/stack/sheet/mineral/plasma/plasma_sheet in rune_turf)
+		resin = new(rune_turf)
+		plasma_sheet.amount--
+		if(plasma_sheet.amount <= 0)
+			qdel(plasma_sheet)
+
+
+//note : some of these can only be built ON resin weeds such as the resin nest.
+GLOBAL_LIST_INIT(resin_recipes, list ( \
+	new/datum/stack_recipe("Resin seed", /obj/structure/alien/weeds/node, 10, one_per_turf = 1, on_floor = 1, time = 100), \
+	new/datum/stack_recipe("Resin Wall", /obj/structure/alien/resin/wall, 3, one_per_turf = 1, on_floor = 1 , time = 50), \
```

Comment:
don't forget about resin windows

Reply:
these are all kinda expensive, seeing as how you need an entire plasma sheet for each resin sheet

either lower the costs of these recipes, or increase amount of resin sheets that you get per plasma sheet